core/trino-main/src/main/java/io/trino/security/AccessControlManager.java

```diff
@@ -200,6 +200,9 @@ public void loadSystemAccessControl(String name, Map<String, String> properties)
             systemAccessControl = factory.create(ImmutableMap.copyOf(properties));
         }
 
+        systemAccessControl.getEventListeners()
+                .forEach(eventListenerManager::addEventListener);
```

Comment:
I would move it to the no-argument `loadSystemAccessControl()`, mostly because this is also where the even listeners are registered for the non-default access controls. This ways it's clearer why this has to be done.

But then you'll have to change `TestingAccessControlManager` to call `loadSystemAccessControl()` instead (and orchestrate all the configuration set-up I presume). Do you think this is feasible?

Reply:
But then we would need to move out `factory.create(..)` from `setSystemAccessControl(String name, Map<String, String> properties)` because without initializing `systemAccessControl`, we can't register event listeners. 

Also in a way `setSystemAccessControl(String name...)` is loading system access control from the name so I think its fine

Reply:
OK then. This was my thought from looking around the code, as I wanted to understand the context of this change.